python/res/enkf/enkf_main.py

```diff
@@ -229,47 +222,17 @@ def _init_res_config(self, config):
         # The res_config argument can be None; the only reason to
         # allow that possibility is to be able to test that the
         # site-config loads correctly.
-        if not config:
-            check = True
-        else:
-            if sys.version_info[0] == 2:
-                check = isinstance(config, basestring)
-            else:
-                check = isinstance(config, str)
-        if check:
-            user_config_file = None
-
-            if sys.version_info[0] == 2:
-               check2 = isinstance(config, basestring)
-            else:
-               check2 = isinstance(config, str)
-            if check2:
-                if not isfile(config):
-                    raise IOError('No such configuration file "%s".' % res_config)
-
-                warnings.warn("Initializing enkf_main with a config_file is "
-                            "deprecated. Please use res_config to load the file "
-                            "instead",
-                            DeprecationWarning
-                            )
-
-                user_config_file = config
-
-            res_config = ResConfig(user_config_file)
-            res_config.convertToCReference(self)
-
-            return res_config
+        if config is None:
+            config = ResConfig(None)
+            config.convertToCReference(self)
+            return config
 
-        raise TypeError("Expected ResConfig, received: %r" % res_config)
+        raise TypeError("Expected ResConfig, received: %r" % config)
```

Comment:
Nitpick, but maybe switch to the new formatting: `raise TypeError("Expected ResConfig, received: {}".format(config))`